qhub/provider/cicd/github.py

```diff
@@ -82,3 +88,219 @@ def create_repository(owner, repo, description, homepage, private=True):
+        env_vars["ARM_TENANT_ID"] = "${{ secrets.ARM_TENANT_ID }}"
+    elif config["provider"] == "do":
+        env_vars["AWS_ACCESS_KEY_ID"] = "${{ secrets.AWS_ACCESS_KEY_ID }}"
+        env_vars["AWS_SECRET_ACCESS_KEY"] = "${{ secrets.AWS_SECRET_ACCESS_KEY }}"
+        env_vars["SPACES_ACCESS_KEY_ID"] = "${{ secrets.SPACES_ACCESS_KEY_ID }}"
+        env_vars["SPACES_SECRET_ACCESS_KEY"] = "${{ secrets.SPACES_SECRET_ACCESS_KEY }}"
+        env_vars["DIGITALOCEAN_TOKEN"] = "${{ secrets.DIGITALOCEAN_TOKEN }}"
+    elif config["provider"] == "gcp":
+        env_vars["GOOGLE_CREDENTIALS"] = "${{ secrets.GOOGLE_CREDENTIALS }}"
+    elif config["provider"] == "local":
+        # create mechanism to allow for extra env vars?
+        pass
+    else:
+        raise ValueError("Cloud Provider configuration not supported")
+
+    return env_vars
+
+
+### GITHUB-ACTIONS SCHEMA ###
+
+
+class GHA_on_extras(BaseModel):
+    branches: List[str]
+    path: List[str]
+
+
+class GHA_on(BaseModel):
+    # to allow for dynamic key names
+    __root__: Dict[str, GHA_on_extras]
+
+    # TODO: validate __root__ values
+    # `push`, `pull_request`, etc.
+
+
+class GHA_job_steps_extras(BaseModel):
+    # to allow for dynamic key names
+    __root__: Union[str, float, int]
+
+
+class GHA_job_step(BaseModel):
+    name: str
+    uses: Optional[str]
+    with_: Optional[Dict[str, GHA_job_steps_extras]] = Field(alias="with")
+    run: Optional[str]
+    env: Optional[Dict[str, GHA_job_steps_extras]]
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class GHA_job_id(BaseModel):
+    name: str
+    runs_on_: str = Field(alias="runs-on")
+    steps: List[GHA_job_step]
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class GHA_jobs(BaseModel):
+    # to allow for dynamic key names
+    __root__: Dict[str, GHA_job_id]
+
+
+class GHA(BaseModel):
+    name: str
+    on: GHA_on
+    env: Optional[Dict[str, str]]
+    jobs: List[GHA_jobs]
+
+
+class QhubOps(GHA):
+    pass
+
+
+class QhubLinter(GHA):
+    pass
+
+
+### GITHUB ACTION WORKFLOWS ###
+
+PYTHON_VERSION = 3.8
+
+
+def checkout_image_step():
+    return GHA_job_step(
+        name="Checkout Image",
+        uses="actions/checkout@master",
+        with_={
+            "token": GHA_job_steps_extras(
+                __root__="{{ '${{ secrets.REPOSITORY_ACCESS_TOKEN }}' }}"
+            )
+        },
+    )
+
+
+def setup_python_step():
+    return GHA_job_step(
+        name="Set up Python",
+        uses="actions/setup-python@v2",
+        with_={"python-version": GHA_job_steps_extras(__root__=PYTHON_VERSION)},
+    )
+
+
+def install_qhub_step(qhub_version):
+    return GHA_job_step(name="Install QHub", run=pip_install_qhub(qhub_version))
+
+
+def gen_qhub_ops(config):
+
+    env_vars = gha_env_vars(config)
+    branch = config["ci_cd"]["branch"]
+    qhub_version = config["qhub_version"]
+
+    push = GHA_on_extras(branches=[branch], path=["qhub-config.yaml"])
```

Comment:
are we considering #995 here?

Reply:
I didn't see this but I think won't be too hard. Do you think we should include it here?

Reply:
I think so, as it was linked to the original issue and it would be just adding an `if` for `step5` based on this [new variable](https://github.com/Quansight/qhub/blob/426c610e042c345c1d2f7ba78aa70595bda0d587/qhub/render/__init__.py#L13)